StyleCop.Analyzers/StyleCop.Analyzers/Helpers/SyntaxTreeHelpers.cs

```diff
@@ -85,16 +85,28 @@ internal static bool ContainsUsingAlias(this SyntaxTree tree, ConcurrentDictiona
                 return result;
             }
 
-            bool generated = ContainsUsingAliasNoCache(tree);
+            bool generated = ContainsUsingAliasNoCache(tree, semanticModel, cancellationToken);
             cache.TryAdd(tree, generated);
             return generated;
         }
 
-        private static bool ContainsUsingAliasNoCache(SyntaxTree tree)
+        private static bool ContainsUsingAliasNoCache(SyntaxTree tree, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
+            // Check for "local" using aliases
             var nodes = tree.GetRoot().DescendantNodes(node => node.IsKind(SyntaxKind.CompilationUnit) || node.IsKind(SyntaxKind.NamespaceDeclaration) || node.IsKind(SyntaxKindEx.FileScopedNamespaceDeclaration));
+            if (nodes.OfType<UsingDirectiveSyntax>().Any(x => x.Alias != null))
+            {
+                return true;
+            }
+
+            // Check for global using aliases
+            if (SemanticModelExtensions.SupportsGetImportScopes)
```

Comment:
In a previous version, I ran either the new or the old code, so I needed this SupportsGetImportScopes method to know which path to take. That turned out not to work. As it is written now, this can be simplified by removing the Supports method, since the fallback of GetImportScopes returns an empty array.

Reply:
Fixed this